### ISSUE TITLE
Not working with docker for windows 

### ISSUE THREAD

**Issue by luissalgadofreire** [LDSSA/setup#3]
I have tried to install following the instructions but the notebook opens with no files inside so the volume mount doesn't seem to work with the given `docker run command` for some reason.

I have the following version of docker for windows (the native one, not the toolbox): `Docker version 17.06.0-ce, build 02c1d87`.

I got it to work using the following equivalent `docker-compose` file though:

```
version: '3.1'

services:

  jupyter:
    restart: always
    image: ldssa/learning-units:latest
    ports:
      - "8888:8888"
    working_dir: /root/units
    volumes:
      - .:/root/units
```

which I run with:
```
docker-compose up -d
```

I don't know why the plain docker command is not mapping the current folder into the container's folder but it's working with this alternative. My concern is just so that others might have a workaround in case their work environment has the same behavior as mine.

**Comment by FrancisBases** [comment]
Hi Luis,
I didn't use Docker and my Anaconda3 Python distro is running smoothly.

Honestly i don't see much interest in using docker because i use Anaconda3 in Win10 as default, it even is accessible in command prompt and Sublime Text.
Check here to see my setup: https://github.com/FrancisBase/setup/wiki 

Can you give more details on how docker is useful to you?

Best,
Franc

**Comment by luissalgadofreire** [comment]
Hi, Francisco.

Don't worry, I have a working environment. I just suggested a workaround in case someone else using a similar environment as mine had the same problem using your alternative docker-based instructions in Windows. I'm not recommending others to follow this method.

I do use docker extensively, though (for every project). It allows me to swiftly setup isolated (and easily reproducible and shareable) working environments per project without meddling with my main system. Easy to setup and destroy, on my laptop or remotely (using docker-machine and docker-swarm against e.g. digitalocean, AWS).

That's not the focus of the academy, though, and I have no intention of pushing this further.

Best,

Luis Freire

**Comment by FrancisBases** [comment]
Ok, thanks for sharing it's always good to know alternatives, it's part of the learning process so your feedback is important also. 

Do you know other alternatives to docker out of curiosity?

Best,

Franc

**Comment by luissalgadofreire** [comment]
I used python [virtualenv](https://pypi.python.org/pypi/virtualenv) for environment isolation at the beginning. Then I started using [vagrant](https://www.vagrantup.com/).  Finally, I bought into docker. It became a cornerstone for modern days devops using the notion of containerization.

Several tools for more complex cluster orchestration (not relevant for single server environments as is the case for us here) have been developed around it (docker swarm, kubernetes, mesos). But docker containers are at the center of those (though alternative container techs have been emerging as alternatives, namely [rocket](https://coreos.com/rkt)).
